src/mongo/db/query/get_executor.cpp

```diff
@@ -306,7 +306,20 @@ void fillOutPlannerParams(OperationContext* opCtx,
         auto collDesc = CollectionShardingState::get(opCtx, canonicalQuery->nss())
                             ->getCollectionDescription(opCtx);
         if (collDesc.isSharded()) {
-            plannerParams->shardKey = collDesc.getKeyPattern();
+            const auto keyPattern = collDesc.getKeyPattern();
```

Comment:
Suggest using `const auto&` here to avoid copying the key pattern object unnecessarily.